### ISSUE TITLE
Unable to create ARM node pools with default settings

### ISSUE THREAD

**Issue by jmthomas** [terraform-google-modules/terraform-google-kubernetes-engine#1437]
### TL;DR

Setting `machine_type = "t2a-standard-2"` throws

    google_container_node_pool.primary_nodes: Creating...
    ╷
    │ Error: error creating NodePool: googleapi: Error 400: Empty name and project for ARM, as current version is not supported., badRequest
    │ 
    │   with google_container_node_pool.primary_nodes,
    │   on gke.tf line 36, in resource "google_container_node_pool" "primary_nodes":
    │   36: resource "google_container_node_pool" "primary_nodes" {

I eventually figured it out by forcing kubernetes version 1.24 and setting the correct zone where the ARM machines exist. Hope this will help others work through similar issues.

### Expected behavior

Expected node pool to be created like other machine types: n1-standard-2, e2-standard-2, etc

### Observed behavior

    google_container_node_pool.primary_nodes: Creating...
    ╷
    │ Error: error creating NodePool: googleapi: Error 400: Empty name and project for ARM, as current version is not supported., badRequest
    │ 
    │   with google_container_node_pool.primary_nodes,
    │   on gke.tf line 36, in resource "google_container_node_pool" "primary_nodes":
    │   36: resource "google_container_node_pool" "primary_nodes" {

### Terraform Configuration

```hcl
region     = "us-central1"
# ARM only available in us-central1-a (and 1-b and 1-f) as of 10/21/22
# per https://cloud.google.com/compute/docs/regions-zones#available
zone = "us-central1-a"
# t2a is an ARM machine for testing via a M2 Macbook
machine_type = "t2a-standard-2"

variable "region" {
  description = "region"
}

variable "zone" {
  description = "zone"
}

provider "google" {
  project = var.project_id
  region  = var.region
}

variable "gke_num_nodes" {
  default     = 2
  description = "number of gke nodes"
}

variable "machine_type" {
  description = "machine type"
}

data "google_container_engine_versions" "location" {
  provider = google
  location = var.zone
  # ARM node pools are only supported in version 1.24
  version_prefix = "1.24."
}

# GKE cluster
resource "google_container_cluster" "primary" {
  name               = "${var.project_id}-gke"
  location           = var.zone
  min_master_version = data.google_container_engine_versions.location.latest_node_version

  # We can't create a cluster with no node pool defined, but we want to only use
  # separately managed node pools. So we create the smallest possible default
  # node pool and immediately delete it.
  remove_default_node_pool = true
  initial_node_count       = 1

  network    = google_compute_network.vpc.name
  subnetwork = google_compute_subnetwork.subnet.name
}

# Separately Managed Node Pool
resource "google_container_node_pool" "primary_nodes" {
  name       = "${google_container_cluster.primary.name}-node-pool"
  location   = var.zone
  cluster    = google_container_cluster.primary.id
  node_count = var.gke_num_nodes
  version    = data.google_container_engine_versions.location.latest_node_version

  node_config {
    oauth_scopes = [
      "https://www.googleapis.com/auth/cloud-platform",
    ]

    labels = {
      env = var.project_id
    }

    # preemptible  = true
    machine_type = var.machine_type
    tags         = ["gke-node", "${var.project_id}-gke"]
    metadata = {
      disable-legacy-endpoints = "true"
    }
  }
}
```

### Terraform Version

```sh
Terraform v1.3.1
on darwin_arm64
```

**Comment by jmthomas** [comment]
It appears you also have to 'untaint' the nodes so they can be scheduled

    kubectl get nodes
    kubectl taint nodes <node name> kubernetes.io/arch=arm64:NoSchedule-

**Comment by bharathkkb** [comment]
Thanks for the report @jmthomas 
Happy to review a PR to add a note on our readme with the requirements pointing to https://cloud.google.com/kubernetes-engine/docs/concepts/arm-on-gke#arm-requirements-limitations

**Comment by anthr76** [comment]
I was able to quickly spin up a arm cluster with the following:


```hcl
module "gke-kutara" {
  source                            = "terraform-google-modules/kubernetes-engine/google"
  version                           = "23.2.0"
  project_id                        = module.kutara-project.project_id
  name                              = "kutara-cluster-arm"
  region                            = "us-central1"
  zones                             = ["us-central1-a", "us-central1-b", "us-central1-f"]
  network                           = module.vpc.network_name
  subnetwork                        = module.vpc.subnets_names[0]
  ip_range_pods                     = "k8s-pod"
  ip_range_services                 = "k8s-service"
  release_channel                   = "RAPID"
  create_service_account            = true
  remove_default_node_pool          = true
  disable_legacy_metadata_endpoints = false

  node_pools = [
    {
      name         = "arm-01"
      min_count    = 1
      max_count    = 3
      spot         = true
      auto_upgrade = true
      autoscaling  = false
      disk_size_gb = 50
      machine_type = "t2a-standard-1"
    }
  ]

  node_pools_metadata = {
    pool-01 = {
      shutdown-script = "kubectl --kubeconfig=/var/lib/kubelet/kubeconfig drain --force=true --ignore-daemonsets=true --delete-local-data \"$HOSTNAME\""
    }
  }

  node_pools_tags = {
    all = [
      "all-nodes",
    ]
    pool-01 = [
      "pool-01-arm64",
    ]
  }

  node_pools_taints = {
    all    = []
    arm-01 = []
  }

}

```


Sadly though I can't make that taint go away...
